sn_client/src/wallet.rs

```diff
@@ -430,7 +462,13 @@ impl WalletClient {
                     .await
                     .map_err(|error| WalletError::CouldNotSendMoney(error.to_string()));
 
-                debug!("Storecosts retrieved for {content_addr:?} {cost:?}");
+                let t = cost
+                    .as_ref()
+                    .expect("Pay for storage once: Cost not found")
```

Comment:
guess here

Reply:
Since this a debug log, I think we can print out the whole `cost` struct here. It contains the details that you've extracted out anyways.
We try not to call `expect/unwrap`, as it would crash the node/client if the value was not present. This is not desirable.

Reply:
I don't think we should be doing `expect` at this layer either. We need to be retuning errors from the `sn_client` lib. `expect` will cause a panic.

Reply:
- I've extracted the expect into a match. I've tried to  [return a specific message from a lib](https://maidsafe.slack.com/archives/C050EDL5ACS/p1707922625611599) but no luck so I've used the linked suggestion from Chris. 

- There is an issue with printing the whole struct. It says the struct does not use the "Debug trait". Even though it's basically the same as the tuple used before. Perhaps I can use a better solution:
In Java I would override the .toString method to print out all the internal values, In Rust perhaps I could concatenate everything into a String to complete the output message? I can do that if necessary.

Reply:
w/r/t `Debug`, I think we can just add the `Debug` trait can we not just add `#[derive(Debug)]` to the struct or its internals?

Reply:
54 | ...rialize, custom_debug::Debug, Debug,
   |             -------------------  ^^^^^ conflicting implementation for `PaymentQuote`

@joshuef  @RolandSherwin 
Ello. So I don't know what the custom debug is for or how to find out; but this is the reason why.
I'm guessing there's a custom debug for a reason, and that I can't change it to debug. It would be scarier if I changed it and nothing happened - so it's probably best to ask 👼